# knowledgebase
General knowledge for HBC computing

* [Standarized templates and packages for the core](http://bioinformatics.sph.harvard.edu/hbcABC/)

# Project set up

[Document](https://docs.google.com/document/d/1FVyPPI7b1KdeDYp7CQRct6Gozpmmy-Xq2P7QeAcxG10/edit?usp=sharing) with suggested procedure on setting up projects [Need to be migrated to MD].

# General tricks

* [OSX](https://github.com/hbc/knowledgebase/blob/master/OSX.md)
* [R-tips-and-tricks](https://github.com/hbc/knowledgebase/blob/master/R-tips-and-tricks.md)
* [Bash](https://github.com/hbc/knowledgebase/blob/master/Bash.md)

# O2

* [helpers to run bcbio in O2](https://code.harvard.edu/HSPH/hbc_bcbio_o2)

# General NGS

* [General NGS](https://github.com/hbc/knowledgebase/blob/master/GeneralNGS.md)

# bcbio
* [tips for running bcbio on HPC](https://github.com/hbc/knowledgebase/blob/master/bcbio_tips.md)
* [tips for genomes](https://github.com/hbc/knowledgebase/blob/master/bcbio_genomes.md)


# Clustering

## External method to test 

If somebody tries and it works, please share a template at hbcACB repo:

* Method to find the best [clusters](https://www.bioconductor.org/help/course-materials/2016/BioC2016/ConcurrentWorkshops1/Risso/clusterExperiment.html) using PCA or any other dimensional reduction algorithm.
* Find significance in your [samples clustering](https://bioconductor.org/packages/release/bioc/vignettes/ClusterSignificance/inst/doc/ClusterSignificance-vignette.html)

# Single-cell RNA-seq Resources

* [SingleCell](https://github.com/hbc/knowledgebase/blob/master/Single-Cell.md)

## External methods

Worthy to check by someone and add it or not to our main documentation:

* [Single Cell Transcriptomics Repo](https://github.com/SingleCellTranscriptomics)
* Tutorial for people starting out with single-cell: 
https://github.com/SingleCellTranscriptomics/ISMB2018_SingleCellTranscriptomeTutorial/wiki
* [Haghverdi et al, 2018](https://europepmc.org/abstract/med/29608177) correcting for confounders/batch by matching mutual nearest neighbors
* [A comparison of single-cell trajectory inference methods](https://www.biorxiv.org/content/early/2018/03/05/276907)
* [Single cell Toolkit uses COMBAT for scRNA-seq batch correction](https://compbiomed.github.io/sctk_docs/articles/v03-tab01_Upload.html)
* [BatchQC](https://bioconductor.org/packages/release/bioc/html/BatchQC.html) Software for assessing best methods for batch correction (based on ComBat methodology). Publication [here](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2263-6)


