# Admin
- [Methods for manuscripts](https://github.com/hbc/methods/blob/master/method_snippets.md)
- [Chargeback models](admin/chargeback_models.md)
- [Data management](data_management.md)
- [Initial consults](initial_consults.md)
- [Migrate to code](migrate_to_code.md)
- [Sequencing download](sequencing_downloads.md)
- [Setting up the analysis](setting_up_an_analysis_guidelines.md)

# bcbio
- https://bcbio-nextgen.readthedocs.io/en/latest/
- https://github.com/bcbio/bcbio-nextgen/issues
- [Bcbio genomes](bcbio/bcbio_genomes.md)
- [bcbio tips](bcbio/bcbio_tips.md)
- [Bcbio multiple files](bcbio/multiple_files_bcbio.md)

# Bulk RNA-seq
- [Count normalizataion methods](https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html)
- [Bibliography](rnaseq/bibliography.md)
- [Dexseq](rnaseq/dexseq.Rmd)
- [Failure types](rnaseq/failure_types)
- [IRFinder_report](rnaseq/IRFinder_report.md)
- [README](rnaseq/README.md)
- [RepEnrich2 guide](rnaseq/RepEnrich2_guide.md)
- [IRFinder](rnaseq/running_IRFinder.md)
- [Strandedness](rnaseq/strandedness.md)
- [Tools](rnaseq/tools.md)

# CHIP-seq
- [ENCODE Guidelines](http://genome.cshlp.org/cgi/pmidlookup?view=long&pmid=22955991) A very useful resource for people starting out with ChIP-seq
- Training materials for ChIP-seq analysis [linked here](https://hbctraining.github.io/Intro-to-ChIPseq/schedule/2-day.html)
- [Tools](chipseq/tools.md)

# How it is organized
- minimal overhead: feel free to add any piece of knowledge in any way
- easy to navigate in one click
- github is tracking changes for you
- it is ok to have multiple similar recipes
- growing naturally
- no placeholders
- please name all pages in small_letters.md

# How to contribute
- nagivate to where you want to share your knowledge or fix something
- don't create a new folder unless you have >=2 new pages that don't fit into existing structure
- add actual value (know how), not structure
- hit 'edit file' or 'new file'
- translate knowledge from your mind into markdown
- hit 'preview changes' to make sure your formatting is fine
- hit 'commit changes'
- thank you, you just have saved somebody's day!

# Knowledge we need to extract (aka TODO):
- ATAC-seq
- miRNA

# Misc
- [Core resources](misc/core_resources.md)
- [General NGS](misc/general_ngs.md)
- [Orphan improvements](misc/orphan_improvements.md)
- [OSX](misc/OSX.md)
- [Snakemake](misc/snakemake-example-pipeline)

# R
- [.Rprofile](Rprofile)
- [RShiny](rshiny_server.md)
- [Tip and tricks](R-tips-and-tricks.md)

# Training team
- HBC Training Github Org: https://github.com/hbctraining
- Upcoming workshops: http://bioinformatics.sph.harvard.edu/training
- Past workshops: http://bioinformatics.sph.harvard.edu/training#past-workshops
- Workshop materials: https://hbctraining.github.io/main
- Training team members:
  - Radhika Khetani
  - Mary Piper
  - Meeta Mistry
  - Jihe Liu
- Training email: [hbctraining (at) hsph.harvard.edu](mailto:hbctraining@hsph.harvard.edu)

# RC (Research Computing, HPC, O2)
- [Connection ot HPC](connection-to-hpc.md)
- [File transfer](file-transfer.md)
- [IPython notebook](ipython-notebook-on-O2.md)
- [Manage files](manage-files.md)
- [O2](O2-tips.md)
- [Scheduler](scheduler.md)

# Single Cell RNA-seq
- [Bcbio indrops3](bcbio_indrops3.md)
- [Readme](README.md)
- [Rstudio docker](rstudio_sc_docker.md)
- [Saturation](scrnaseq/saturation_qc.md)
- [Clustering analysis in Seurat](scrnaseq/seurat_clustering_analysis.md)
- [Seurat markers](scrnaseq/seurat_markers.md)
- [Single Cell conda](scrnaseq/Single-Cell-conda.md)
- [Single cell](scrnaseq/Single-Cell.md)
- [Tinyatlas](scrnaseq/tinyatlas.md)
- [Tools](scrnaseq/tools.md)
- [Tutorials](scrnaseq/tutorials.md)

# WGS
- [CRISPR offtarget](wgs/crispr-offtarget.md)
- [Pacbio genome assmelby](wgs/pacbio_genome_assembly.md)
